### DISCUSSION TITLE
Step 3 returns unauthorized client

### DISCUSSION THREAD

**Discussion by nikemeal** [timdorr/tesla-api#408] [Q&A] [answered]
Finally got my code to reach step 3 and getting the bearer token.  I pass over the query parameters as suggested:
```JSON
{
  "grant_type": "authorization_code",
  "client_id": "ownerapi",
  "code": "my_auth_code",
  "code_verifier": "my_code_verifier",
  "redirect_uri": "https://auth.tesla.com/void/callback"
}
```
and i get back 400 Unauthorized client with the message "the client id is not authorized".  
I'm assuming it relates to "ownerapi" as the oauth client id is only passed in the step after this.  

Has anyone hit anything like this and if so what was the way through?

**Comment by nikemeal** [comment, marked as answer]
ah, the docs say the fields above are 'request parameters' so I assumed it meant send then as the query params like the initial requests.  But they should be sent as actual POST body payload fields.  My bad

**Reply by Programistich** [reply]
Can u explain what ur problem was?
